exo-go/src/application/docker_compose_builder.go

```diff
@@ -14,18 +14,31 @@ import (
 // DockerComposeBuilder returns the given service config that will appear
 // in docker-compose.yml
 type DockerComposeBuilder struct {
-	AppConfig     types.AppConfig
-	ServiceConfig types.ServiceConfig
-	ServiceData   types.ServiceData
-	Role          string
-	AppDir        string
-	HomeDir       string
+	AppConfig                types.AppConfig
+	ServiceConfig            types.ServiceConfig
+	ServiceData              types.ServiceData
+	BuiltAppDependencies     map[string]config.AppDependency
```

Comment:
I agree that is not a good variable name but this is tricky because we have `types.Dependencies` and `types.Dependency` which refer to the fields in `types.AppConfig`. `config.AppDepency` refers the type that has all functions we need for compiling docker config. There are [cases](https://github.com/Originate/exosphere/blob/e4696af7e6ce49b7c9176f2a952594dd22ccff4a/exo-go/src/application/docker_compose_builder.go#L39) where I need to iterate through both `appConfig.Dependencies` or `serviceConfig.Dependencies` and `builtDependencies` so having different names avoids confusion.